pybamm/solvers/solution.py

```diff
@@ -70,6 +70,10 @@ def __init__(
         for time in t:
             self._known_evals[time] = {}
 
+        # Copy the timescale_eval and lengthscale_evals
+        self.timescale_eval = self._model.timescale_eval
+        self.length_scales_eval = self._model.length_scales_eval
+
```

Comment:
I guess this is the bit that required you to define `model.timescale_eval` at the start? I think we just need to be a bit more careful here. Maybe these two lines could be:
```python3
if hasattr(self._model, "timescale_eval"):
         self.timescale_eval = self._model.timescale_eval
else:
         self.timescale_eval =self._model.timescale.evaluate()
```
or even better put that logic into a class property `model.timescale_eval`

Reply:
I've changed this